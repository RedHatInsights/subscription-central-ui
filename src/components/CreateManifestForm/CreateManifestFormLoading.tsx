import React, { FC } from 'react';
import {
  EmptyState,
  EmptyStateBody,
  EmptyStateVariant,
  EmptyStateHeader
} from '@patternfly/react-core';
import { Processing } from '../emptyState';

interface CreateManifestFormLoadingProps {
  title: string;
}
const CreateManifestFormLoading: FC<CreateManifestFormLoadingProps> = ({ title }) => {
  return (
    <EmptyState variant={EmptyStateVariant.sm}>
      <EmptyStateHeader titleText={<>{title}</>} headingLevel="h3" />
      <EmptyStateBody>
        <Processing />
      </EmptyStateBody>
    </EmptyState>
  );
};

export default CreateManifestFormLoading;
