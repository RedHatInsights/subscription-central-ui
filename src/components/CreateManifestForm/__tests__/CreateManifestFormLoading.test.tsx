import React from 'react';
import { render } from '@testing-library/react';
import CreateManifestFormLoading from '../CreateManifestFormLoading';

describe('Create Manifest Form Loading Message', () => {
  it('renders correctly', () => {
    const { container } = render(<CreateManifestFormLoading title="Loading..." />);
    expect(container).toMatchSnapshot();
  });
});
