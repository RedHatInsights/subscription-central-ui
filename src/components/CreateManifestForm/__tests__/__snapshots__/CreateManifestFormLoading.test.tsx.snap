// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Manifest Form Loading Message renders correctly 1`] = `
<div>
  <div
    class="pf-c-empty-state pf-m-sm"
  >
    <div
      class="pf-c-empty-state__content"
    >
      <h3
        class="pf-c-title pf-m-lg"
      >
        Loading...
      </h3>
      <div
        class="pf-c-empty-state__body"
      >
        <div
          class="pf-l-bullseye"
        >
          <span
            aria-valuetext="Loading..."
            class="pf-c-spinner pf-m-xl"
            role="progressbar"
          >
            <span
              class="pf-c-spinner__clipper"
            />
            <span
              class="pf-c-spinner__lead-ball"
            />
            <span
              class="pf-c-spinner__tail-ball"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
